spacy/lang/nl/punctuation.py

```diff
@@ -0,0 +1,39 @@
+# coding: utf8
+from __future__ import unicode_literals
+
+from ..char_classes import LIST_ELLIPSES, LIST_ICONS
+from ..char_classes import CONCAT_QUOTES, ALPHA, ALPHA_LOWER, ALPHA_UPPER
+from ..punctuation import TOKENIZER_SUFFIXES as DEFAULT_TOKENIZER_SUFFIXES
+
+
+# Copied from `de` package. Main purpose is to ensure that hyphens are not
+# split on.
+_quotes = CONCAT_QUOTES.replace("'", "")
+
+_infixes = (
+    LIST_ELLIPSES
+    + LIST_ICONS
+    + [
+        r"(?<=[{al}])\.(?=[{au}])".format(al=ALPHA_LOWER, au=ALPHA_UPPER),
+        r"(?<=[{a}])[,!?](?=[{a}])".format(a=ALPHA),
+        r"(?<=[{a}])[:<>=](?=[{a}])".format(a=ALPHA),
+        r"(?<=[{a}]),(?=[{a}])".format(a=ALPHA),
+        r"(?<=[{a}])([{q}\)\]\(\[])(?=[{a}])".format(a=ALPHA, q=_quotes),
+        r"(?<=[{a}])--(?=[{a}])".format(a=ALPHA),
+        r"(?<=[0-9])-(?=[0-9])",
```

Comment:
Would it make sense to use `char_classes.HYPHENS` here instead of only `-` ?

Reply:
Wasn't aware that there was a character class for that. Seems like that would make sense!

Reply:
So, I've just taken a closer look at this. Turns out some caution is required when using the `HYPHENS` class, depending on whether it's hyphens (`-`) or dashes (`–` (em-dash), `—`  (en-dash), `--`, `---`, etc.)  that you want to split or, by exclusion, preserve. This is because `HYPHENS` conflates hyphens and dashes into a single set. 

In the case of the pattern on line 23, it makes sense to use the `HYPHENS` character class because our goal is to split on both hyphens and dashes.

In the case of the penultimate pattern on line 22, however, we should use a variant of `HYPHENS` that (confusingly enough) leaves out the actual hyphen character -- the effect being that hyphens are *not* split on. Otherwise, compound words like `zwart-witconclusie` are tokenized as `['zwart', '-', 'witconclusie']`.

I'm having some trouble getting the regular expressions to compile though. Is this potentially something you would like to take a closer look at, @svlandeg?

Reply:
You might have to introduce a new variable in `char_classes.py`:
`CONCAT_HYPHENS = group_chars(_hyphens)`
and then do something like 
`r"(?<=[0-9])[{h}](?=[0-9])".format(h=CONCAT_HYPHENS)`

For line 22 you can make a custom variant of `CONCAT_HYPHENS ` (which I would personally define in the `nl` file because it's lang-specific)

I haven't checked/run this code yet though, as I'm in the middle of a different PR ;-)